scripts/docker_push.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# Dockerhub login
+[ ! -z "$DOCKER_PASSWORD" ] || [ ! -z "$DOCKER_USERNAME" ];
+  echo "DOCKER_USERNAME and DOCKER_PASSWORD must be set as environment variables!";
+  exit 1;
+echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+
+# if the build is for a pushed tag, use that tag for the Docker image;
+# otherwise, use "latest"
+if [[ -z "$TRAVIS_TAG" ]]; then
```

Comment:
You can use Bash's [parameter substitution](https://www.tldp.org/LDP/abs/html/parameter-substitution.html) for `${var:-default_value}` here to greatly simplify the following lines:
`image_release_tag="${TRAVIS_TAG:-latest}"`

Reply:
Wow, this is a beautiful simplification!